### DISCUSSION TITLE
Having IPython Session in VS Code

### DISCUSSION THREAD

**Discussion by RoyiAvital** [winpython/winpython#976] [Q&A]
In VS Code there is a terminal which runs the Python scripts.
I was wondering, I'd like to have IPython session in the terminal. Is there a way? Could the IPython session share the same variables as the script?

**Comment by stonebig** [comment]
in latest VSCode, you can see an "run in interactive Window"... that is probably the prototype of what you want

but there are maybe better option in the StackOverflow thread 
https://stackoverflow.com/questions/52310689/use-ipython-repl-in-vs-code#:~:text=Type%20Ipython%20inside%20the%20terminal,in%20the%20Ipython%20terminal%20window.

![image](https://user-images.githubusercontent.com/4312421/115990789-84cb7a00-a5c5-11eb-989a-9f5c34e7b188.png)

**Reply by RoyiAvital** [reply]
This exactly what I wanted. To run IPython. Just you don't have a launcher for regular IPython REPL. You only have the launcher for QT based one.

**Reply by RoyiAvital** [reply]
@stonebig , What's the correct launcher of IPython in WinPython?

**Reply by stonebig** [reply]
Don't know... what is the answer on internet for this question ? Maybe we miss one

**Reply by RoyiAvital** [reply]
There is the `ipython.exe`. It just requires portable launcher as you created to other binaries.

**Comment by stonebig** [comment]
also the default ptpython option under current "WinPython Interpreter" icon allows you to switch to ipython mode:
![image](https://user-images.githubusercontent.com/4312421/117541720-7a719d00-b015-11eb-92a9-87561e1c23b2.png)
